metron-interface/metron-rest/src/test/java/org/apache/metron/rest/controller/SearchControllerIntegrationTest.java

```diff
@@ -236,17 +236,14 @@ public void test() throws Exception {
             .andExpect(jsonPath("$.groupResults[0].groupResults[0].score").value(50));
 
     this.mockMvc.perform(post(searchUrl + "/column/metadata").with(httpBasic(user, password)).with(csrf()).contentType(MediaType.parseMediaType("application/json;charset=UTF-8")).content("[\"bro\",\"snort\"]"))
-            .andExpect(status().isOk())
-            .andExpect(content().contentType(MediaType.parseMediaType("application/json;charset=UTF-8")))
-            .andExpect(jsonPath("$.*", hasSize(2)))
-            .andExpect(jsonPath("$.bro.common_string_field").value("string"))
-            .andExpect(jsonPath("$.bro.common_integer_field").value("integer"))
-            .andExpect(jsonPath("$.bro.bro_field").value("boolean"))
-            .andExpect(jsonPath("$.bro.duplicate_field").value("date"))
-            .andExpect(jsonPath("$.snort.common_string_field").value("string"))
-            .andExpect(jsonPath("$.snort.common_integer_field").value("integer"))
-            .andExpect(jsonPath("$.snort.snort_field").value("double"))
-            .andExpect(jsonPath("$.snort.duplicate_field").value("long"));
+        .andExpect(status().isOk())
```

Comment:
Does the `SearchControllerIntegrationTest` actually exercise the ElasticsearchDAO?  I think this test will only drive the `InMemoryDao`. 

Please correct me if I am wrong, I am basing this off memory, but I don't believe it actually uses the ElasticsearchDAO.  If I am right, we probably need to update the integration tests that drive the ElasticsearchDAO.

Reply:
And I should add, you did update the `SearchIntegrationTest` to test your changes to `ElasticsearchDAO`.  So I am happy with the level of testing you have.

The test changes here really only test the `InMemoryDao`, I think.  I find the purpose of some of the tests here slightly confusing because they are not really testing the `ElasticsearchDAO`, but that is a side issue that doesn't matter all that much.

Reply:
Yes the `SearchControllerIntegrationTest` does not actually exercise the ElasticsearchDAO and I find this confusing as well.  Really all the `SearchControllerIntegrationTest` tests is that serialization/deserialization happens correctly (which is still valuable).

Reply:
Just some historical color to this discussion.  metron-rest's tests do not assume ES, rather the ES Dao tests are in metron-elasticsearch.  The idea was that metron-rest should have a runtime dependency on the index being used, rather than a compile-time dependency.  The thought was that the testing for the controller and such should assume an implementation of the DAO that conforms to the interface allowing the ES DAO tests to ensure that it's conforming to the expectations of the interface independently.